src/main/java/com/bobocode/svydovets/annotation/bean/processor/AutoSvydovetsBeanPostProcessor.java

```diff
@@ -18,26 +19,31 @@ public AutoSvydovetsBeanPostProcessor() {
         this.beanFactory = new AnnotationBeanFactory();
     }
 
+    public AutoSvydovetsBeanPostProcessor(BeanFactory beanFactory) {
+        this.beanFactory = beanFactory;
+    }
+
+
     @Override
     public void processBeans(Map<String, Object> rootContext) {
-        for (var entrySet : rootContext.entrySet()) {
-            Object beanObject = entrySet.getValue();
+        for (var entry : rootContext.entrySet()) {
+            Object beanObject = entry.getValue();
             Class<?> beanType = beanObject.getClass();
             for (var field : beanType.getDeclaredFields()) {
                 if (field.isAnnotationPresent(AutoSvydovets.class)) {
-                    var dependency = beanFactory.getBean(field.getType());
+                    var dependency = getDependencyForField(field);
```

Comment:
Not a real comment,
We need to clarify how we will handle other injection approaches in code (By constructor and method). This information can be stored in bean definition ?

Reply:
I was thinking about same approach as for fields
```
for (var constructor : beanType.getDeclaredConstructors())
...
for (var method : beanType.getDeclaredMethods())
...
```